pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -80,6 +94,11 @@ func (a *Authenticator) AuthenticatePassword(username, password string) (user.In
 
 	resp, err := a.client.Do(req)
 	if err != nil {
+		if _, ok := err.(*RedirectAttemptedError); !ok {
+			// If this was a redirect, treat it as a simple
+			// authentication failure, not an error
```

Comment:
The 500 error is returned by `tokencmd.RequestToken()` to the user. I'm not sure how exactly it behaves in the web UI yet.

So how do you want me to proceed here?